src/json_struct.rs

```diff
@@ -4,178 +4,421 @@ use std::collections::btree_map::Iter as MapIter;
 use std::slice::Iter as VecIter;
 
 use rustc_serialize::json::Json;
+use serde_json::value::Value;
+
+pub trait ToGron {
+    fn to_gron<W: Write>(&self, out: &mut W, prefix: &str) -> io::Result<()>;
+}
+
+pub fn json_to_gron<W: Write, T: ToGron>(out: &mut W, prefix: &str, json: &T)
+    -> io::Result<()>
+{
+    json.to_gron(out, prefix)
+}
 
 
 enum StackItem<'a> {
-    MapIter(MapIter<'a, String, Json>),
-    VecIter(Enumerate<VecIter<'a, Json>>),
+    MapIterSerial(MapIter<'a, String, Json>),
```

Comment:
This shouldn't be a single structure. The items of serde and RustcSerialize are mutually exclusive.